azkaban-common/src/main/java/azkaban/utils/EmailMessage.java

```diff
@@ -267,8 +267,8 @@ private void retrySendMessage(final SMTPTransport t, final Message message)
       try {
         t.sendMessage(message, message.getRecipients(Message.RecipientType.TO));
         return;
-      } catch (final MessagingException sme) {
-        this.logger.error("Sending email messages failed, attempt: " + attempt, sme);
+      } catch (final Exception e) {
+        this.logger.error("Sending email messages failed, attempt: " + attempt, e);
```

Comment:
more clear logging: "Sending email messages failed, attempt: " + (attempt+1)

Reply:
should we have a sleep interval between each retry?

Reply:
There are sleep intervals for mailTimeout and ConnectionTimeout:
    final int mailTimeout = props.getInt("mail.timeout.millis", 30000);
    final int connectionTimeout =
        props.getInt("mail.connection.timeout.millis", 30000);